### PR TITLE
feat: added a namingconvention optional parameter to the Nhost data p…

### LINKED ISSUE #4988: [FEAT] Nhost dataProvider requires aggregateOperation name paramater in meta
### Is your feature request related to a problem? Please describe.

I would like to display Nhost users in an Antd table. Nhost predefines and uses camelCase i.e.,`usersAggregate` for the Select Aggregate on the users table. This is not compatible with the snake_case operation names found throughout Hasura and the dataProvider useList function
```
const aggregateOperation = defaultNamingConvention
                ? `${operation}_aggregate`
                : camelCase(`${operation}_aggregate`);
```

changing the select aggregate in Hasura directly breaks the Nhost console auth section as it relies on the name to be `usersAggregate`

### Describe alternatives you've considered

the operation name can be customised with 
```const operation = defaultNamingConvention
                ? meta?.operation ?? resource
                : camelCase(meta?.operation ?? resource);
```
so, something like this maybe? 
```
const aggregateOperation = defaultNamingConvention
                ? meta?.aggregateOperation ?? `${operation}_aggregate`
                : camelCase(`${operation}_aggregate`);
```

### Additional context

Currently using a customised dataProvider that returns  
```
{
    ...dataProvider(client),
     getList: the modified function that allows meta?.aggregateOperation
}
```

### Describe the thing to improve

I would like to be able to query the user list with a custom aggregate operation name

### PR DESCRIPTION
**Pull request by Ponnada96**
feat: added a namingConvention optional parameter to the Nhost data provider 

closes #4988

### Self Check before Merge

Please check all items below before review.

-   [x] Corresponding issues are created/updated or not needed
-   [x] Docs are updated/provided or not needed
-   [x] Examples are updated/provided or not needed
-   [x] TypeScript definitions are updated/provided or not needed
-   [x] Tests are updated/provided or not needed
-   [x] Changesets are provided or not needed

### CODE REVIEW

File: packages/nhost/src/dataProvider/index.ts [hunk 1]
```diff
@@ -57,34 +58,34 @@ export type HasuraFilterCondition =
...
-        endswith: "_iregex",
-        nendswith: "_iregex",
-        startswiths: "_similar",
-        nstartswiths: "_nsimilar",
-        endswiths: "_similar",
-        nendswiths: "_nsimilar",
-    };
+{
+    eq: "_eq",
+    ne: "_neq",
```
Comment (BatuhanW) [inline packages/nhost/src/dataProvider/index.ts:63]: Let's revert format changes here as they aren't relevant.

File: packages/nhost/src/dataProvider/index.ts [hunk 2]
```diff
@@ -159,15 +160,20 @@ const handleError = (error: object | Error) => {
...
 };
 
 const dataProvider = (
     client: NhostClient,
     options?: NHostDataProviderOptions,
 ): Required<DataProvider> => {
-    const { idType } = options ?? {};
+    const { idType, namingConvention = 'hasura-default' } = options ?? {};
+    const defaultNamingConvention = namingConvention === 'hasura-default'
+
```
Comment (BatuhanW) [inline packages/nhost/src/dataProvider/index.ts:176]: Could you add/update tests?

**Review by BatuhanW — COMMENTED** [review]
Hello @Ponnada96 thanks for the PR!

Added some comments, also could you add patch changeset when you finish PR?

https://refine.dev/docs/contributing/#changeset

File: packages/nhost/src/dataProvider/index.ts (hunk 1 above)
Comment (Ponnada96) [reply packages/nhost/src/dataProvider/index.ts:63]: yes sure, I will revert the  format changes

File: packages/nhost/src/dataProvider/index.ts (hunk 2 above)
Comment (Ponnada96) [reply packages/nhost/src/dataProvider/index.ts:176]: Yes Sure

**Pull request closed without merging**